### PR TITLE
v1 of set draft format

### LINKED ISSUE #1512: Working v1 prototype of 'Set Draft' format

### LINKED ISSUE #1514: Disallow set formats in Matchmaking mode (because too unlikely to find a match)

### PR DESCRIPTION
**Pull request by AlexNisnevich**
[ close #1512, close #1514 ]

Note that a big chunk of this PR is factoring `GameAreaContents` and `LeftControls` out of `GameArea` (to facilitate setting up the new `DraftArea` interface).

image

### CODE REVIEW

File: src/common/components/game/DraftDeck.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+interface DraftDeckProps {
+  cards: w.CardInGame[]
+}
+
+export default class DraftDeck extends React.PureComponent<DraftDeckProps> {
+  public render(): JSX.Element {
+    const { cards } = this.props;
+
+    return (
+      <div style={{  }}>
```
Comment (jacobnisnevich) [inline src/common/components/game/DraftDeck.tsx:17]: Delete this?

File: src/common/components/game/DraftDeck.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        {
+          sortBy(cards, ['cost', 'name']).map((card, idx) =>
+            <div
+              key={idx}
+              style={{ float: 'left' }}
```
Comment (jacobnisnevich) [inline src/common/components/game/DraftDeck.tsx:22]: Why float over flex?

File: src/common/components/game/DraftDeck.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            >
+              {Card.fromObj(card, { scale: 0.6 })}
```
Comment (jacobnisnevich) [inline src/common/components/game/DraftDeck.tsx:24]: Maybe pull out 0.6 as a constant?

**Review by jacobnisnevich — APPROVED** [review]
Couple nits, mostly lgtm though

**Review by Dbz — COMMENTED** [review]
Awesome!

File: src/common/components/game/DraftDeck.tsx (hunk 2 above)
Comment (AlexNisnevich) [reply src/common/components/game/DraftDeck.tsx:22]: I still to this day don't understand flex

**Pull request merged**